packages/util/src/bytes.ts

```diff
@@ -300,3 +306,29 @@ export const validateNoLeadingZeroes = function (values: { [key: string]: Buffer
     }
   }
 }
+
+/**
+ * Converts a {@link Uint8Array} or {@link NestedUint8Array} to {@link Buffer} or {@link NestedBufferArray}
+ */
+export function arrToBufArr(arr: Uint8Array): Buffer
+export function arrToBufArr(arr: NestedUint8Array): NestedBufferArray
+export function arrToBufArr(arr: Uint8Array | NestedUint8Array): Buffer | NestedBufferArray
+export function arrToBufArr(arr: Uint8Array | NestedUint8Array): Buffer | NestedBufferArray {
```

Comment:
I'm not quite sure why I needed to do this final line twice, I thought I was getting pretty good at understanding these overloads too hehe, maybe a TypeScript wiz can help inform me, but I believe I set it up correctly.

Reply:
Actually, instead of doing this, it would be much cooler if the return type was the same shape array as the input type but Buffers replaced with Uint8Arrays. That would probably require much less casting afterward. I'm not sure if this can be accomplished with generics, but I'll read the documentation and try to see.